apps/src/gamelab/spritelab/spriteUtils.js

```diff
@@ -0,0 +1,218 @@
+var spriteId = 0;
+var nativeSpriteMap = {};
+var inputEvents = [];
+var behaviors = [];
+
+export var background;
+export var title = '';
+export var subtitle = '';
+
+export function reset() {
+  spriteId = 0;
+  nativeSpriteMap = {};
+  inputEvents = [];
+  behaviors = [];
+  background = 'white';
+  title = subtitle = '';
+}
+
+/**
+ * Returns a list of all sprites that have the specified animation.
+ * Called on each tick of the draw loop because animations can change throughout runtime.
+ * @param {string} animation - animation name
+ */
+function allSpritesWithAnimation(animation) {
+  let group = [];
+  Object.keys(nativeSpriteMap).forEach(spriteId => {
+    if (nativeSpriteMap[spriteId].getAnimationLabel() === animation) {
+      let sprite = nativeSpriteMap[spriteId];
+      if (sprite) {
+        group.push(sprite);
+      }
+    }
+  });
+  return group;
+}
+
+/**
+ * Returns a list of sprites, specified either by id or animation name.
+ * @param {(string|number)} spriteOrGroup - Either the id or the animation name
+ * @return {[Sprite]} List of sprites that match the parameter. Either a list containing the one sprite
+ * the specified id, or a list containing all sprites with the specified animation.
+ */
+export function singleOrGroup(spriteOrGroup) {
+  if (typeof spriteOrGroup === 'number') {
+    const sprite = nativeSpriteMap[spriteOrGroup];
+    if (sprite) {
+      return [sprite];
+    }
+  }
+  if (typeof spriteOrGroup === 'string') {
+    return allSpritesWithAnimation(spriteOrGroup);
+  }
+  return [];
+}
+
+export function getAnimationsInUse() {
+  let animations = new Set();
+  Object.keys(nativeSpriteMap).forEach(spriteId => {
+    animations.add(nativeSpriteMap[spriteId].getAnimationLabel());
+  });
+  return Array.from(animations);
+}
+
+export function getBehaviorsForAnimation(animation) {
```

Comment:
Perhaps rename or add a comment. I was thinking this would return a list of behaviors for an animation. But it returns a count of animations that have behaviors.

Reply:
On that note, it seems like "Costume" would be more clear than "Animation." Do you know why we use "Animation" for spritelab?

Reply:
"Animation" is the p5 name, and "Costume" is the product name for animations in sprite lab. Chris suggested using "animation" throughout so that if we someday change what we're calling them in spritelab, it won't require code changes here

Reply:
Renaming this function would require changing all the level validation code. I'm happy to do it, but I'd rather do it as a follow-on task once this PR is merged. I'll add a comment for now.

Reply:
Sounds fine to me. I'm ok with leaving it as-is with a comment.